src/App.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState, useCallback, useRef } from "react";
+import "./App.css";
+import { secretText, keyBetweenTime, showDataTime } from "./Common/global";
+import Header from "./components/Header";
+import Main from "./components/Main";
+import { blogType } from "./Common/global";
+
+function App() {
+  const intervalRef: { current: NodeJS.Timeout | null } = useRef(null);
```

Comment:
useRef<number | null>(null) might be okay. Even if you want to use NodeJS.Timeout, 
```
const intervalRef = useRef<NodeJS.Timeout | null>(null)
```
should be preferred.